### PR TITLE
Hardcoded typesupport for Fibonacci action

### PR DESCRIPTION
**Pull request by sloretz**
Split from #44. This PR has hardcoded type support classes for a Fibonacci action.

requires #44

### CODE REVIEW

File: test_msgs/include/test_msgs/action/fibonacci.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#include <test_msgs/action/fibonacci__feedback.h>
+#include <test_msgs/action/fibonacci__goal.h>
+#include <test_msgs/action/fibonacci__result.h>
+#include "test_msgs/action/fibonacci__type_support.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+#define TEST_MSGS__ACTION__FIBONACCI_H_
```
Comment (hidmic) [inline test_msgs/include/test_msgs/action/fibonacci.h]: @sloretz `#endif  // TEST_MSGS__ACTION__FIBONACCI_H_ ` ?

**Review by hidmic — APPROVED** [review]
LGTM! Let's get this going.

**Comment by hidmic** [comment]
Also, it looks like this branch could use a rebase.

**Review by jacobperron — APPROVED** [review]
Other than the one comment from @hidmic, this looks good to go :+1:

File: test_msgs/include/test_msgs/action/fibonacci.h (hunk 1 above)
Comment (sloretz) [reply test_msgs/include/test_msgs/action/fibonacci.h]: Fixed in f99efcfaa4e06d38569cafef87a46b33bdf39457

**Review by jacobperron — APPROVED** [review]

**Comment by sloretz** [comment]
CI
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5500/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2168/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4540/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5402/)

PRs in CI

* https://github.com/ros2/rosidl_dds/pull/45
* https://github.com/ros2/rosidl_dds/pull/45
* https://github.com/ros2/rosidl_python/pull/17
* https://github.com/ros2/rosidl_typesupport_connext/pull/13
* https://github.com/ros2/rosidl_typesupport_opensplice/pull/8
* https://github.com/ros2/rosidl_typesupport_fastrtps/pull/12
* https://github.com/ros2/rosidl_typesupport/pull/35
* https://github.com/ros2/rosidl/pull/310

**Comment by sloretz** [comment]
Checking if Windows dll linkage issue is fixed by a4dfb38fc14dd867196aa3417a020406f832a05c

[Build Status](https://ci.ros2.org/job/ci_windows/5409/)

**Comment by sloretz** [comment]
Checking if WIndows dll linkage issue is fixed by 6d6da5801ed3ee1438246bc5b31898d12d522ff0

[Build Status](https://ci.ros2.org/job/ci_windows/5410/)

**Comment by sloretz** [comment]
Rerunning full CI now that windows linker issue is fixed

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5507/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2175/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4547/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5411/)

**Comment by hidmic** [comment]
@sloretz looks like this is ready!!

**Comment by sloretz** [comment]
@hidmic I think it's ready. All that is left is for ros2/rosidl_python#17 and ros2/rosidl#310 to pass code review and possibly another round of CI after fixing issues found during their reviews.

**Comment by mjcarroll** [comment]
@sloretz This appears to break `test_msgs` on:

* `packaging_osx` : https://ci.ros2.org/view/packaging/job/packaging_osx/1230/
* `packaging_windows`: https://ci.ros2.org/view/packaging/job/packaging_windows/1181/
* `packaging_windows_debug`: https://ci.ros2.org/view/packaging/job/packaging_windows_debug/126/

```
10:12:19 Undefined symbols for architecture x86_64:
10:12:19   "_rosidl_typesupport_opensplice_c__get_message_type_support_handle__test_msgs__srv__Fibonacci_Goal_Request", referenced from:
10:12:19       _send_request__action__Fibonacci_Goal in fibonacci__goal__type_support_c.cpp.o
10:12:19       _take_request__action__Fibonacci_Goal in fibonacci__goal__type_support_c.cpp.o
10:12:19   "_rosidl_typesupport_opensplice_c__get_message_type_support_handle__test_msgs__srv__Fibonacci_Goal_Response", referenced from:
10:12:19       _send_response__action__Fibonacci_Goal in fibonacci__goal__type_support_c.cpp.o
10:12:19       _take_response__action__Fibonacci_Goal in fibonacci__goal__type_support_c.cpp.o
10:12:19   "_rosidl_typesupport_opensplice_c__get_message_type_support_handle__test_msgs__srv__Fibonacci_Result_Request", referenced from:
10:12:19       _send_request__action__Fibonacci_Result in fibonacci__result__type_support_c.cpp.o
10:12:19       _take_request__action__Fibonacci_Result in fibonacci__result__type_support_c.cpp.o
10:12:19   "_rosidl_typesupport_opensplice_c__get_message_type_support_handle__test_msgs__srv__Fibonacci_Result_Response", referenced from:
10:12:19       _send_response__action__Fibonacci_Result in fibonacci__result__type_support_c.cpp.o
10:12:19       _take_response__action__Fibonacci_Result in fibonacci__result__type_support_c.cpp.o
10:12:19 ld: symbol(s) not found for architecture x86_64
```

**Pull request merged**